spinalcordtoolbox/moco.py

```diff
@@ -966,18 +931,49 @@ def split_to_odd_and_even(param):
             im_data_merged.data[:, :, index, :] = im_data_odd.data[:, :, counter_odd, :]
             counter_odd += 1
 
-    im_data_merged.save(file_data_merged, verbose=0)  # Save to /tmp dir
+    im_data_merged.save(sct.add_suffix(file_data, '_merged'), verbose=0)  # Save to /tmp dir
 
-    # TODO - move data from /tmp to current dir - lines below save to current dir original image instead of image
-    #  after moco - fix it
     # Generate output files
     sct.printv('\nGenerate output files...', param.verbose)
-    fname_moco = os.path.join(path_out_abs, sct.add_suffix(os.path.basename(orig_name), param.suffix))
-    sct.generate_output_file(im_data_merged.absolutepath, fname_moco)
+    fname_moco = os.path.join(path_out_abs, sct.add_suffix(os.path.basename(orig_name), '_moco'))
+    #sct.generate_output_file(im_data_merged.absolutepath, fname_moco)
+    sct.generate_output_file(os.path.join(os.getcwd(),sct.add_suffix(file_data, '_merged')), fname_moco)
 
     # come back to working directory
     os.chdir(curdir)
 
     # display elapsed time
     elapsed_time = time.time() - start_time
     sct.printv('\nFinished! Elapsed time: ' + str(int(np.round(elapsed_time))) + 's', param.verbose)
+
+def split_to_even_and_odd(data_to_split, file_name):
+    """
+    Split 4D data along SI direction into two datasets (even and odd slices) and save them
+    :param data_to_split: 4D data to split
+    :param file_name: data filename
+    :return: None
+    """
+    # change orientation to RPI
+    orig_orientation = data_to_split.orientation
+    data_to_split.change_orientation(orientation='RPI',generate_path=True)
+    # split along SI direction
+    im_data_split_list = split_data(data_to_split, 2)
+
+    # Get only even slices across all volumes
+    data_even = []
+    # loop across even slices
+    for index_even in range(0, len(im_data_split_list), 2):
+        data_even.append(im_data_split_list[index_even])
+    # Concatenate in SI and save
+    concat_data(data_even, dim=2).save(sct.add_suffix(file_name, '_even'), verbose=0)
+
+    # Get only odd slices across all volumes
+    data_odd = []
+    # loop across odd slices
+    for index_odd in range(1, len(im_data_split_list), 2):
+        data_odd.append(im_data_split_list[index_odd])
+    # Concatenate in SI and save
+    concat_data(data_odd, dim=2).save(sct.add_suffix(file_name, '_odd'), verbose=0)
+
+    # TODO - reorient data back if they were not RPI
```

Comment:
right, this needs to be done (because down the line, the `moco_wrapper` treats axially- and sagittally-acquired data differently